src/main/java/org/springframework/data/elasticsearch/core/MappingBuilder.java

```diff
@@ -83,6 +92,18 @@
 	static XContentBuilder buildMapping(Class clazz, String indexType, String idFieldName, String parentType) throws IOException {
 
 		XContentBuilder mapping = jsonBuilder().startObject().startObject(indexType);
+
+		// Dynamic templates
+		if (clazz.isAnnotationPresent(DynamicTemplates.class)){
+			String mappingPath = ((DynamicTemplates) clazz.getAnnotation(DynamicTemplates.class)).mappingPath();
+			if (isNotBlank(mappingPath)) {
+				String jsonString = ElasticsearchTemplate.readFileFromClasspath(mappingPath);
+				if (isNotBlank(jsonString)) {
+					addDynamicTemplatesMapping(mapping, jsonString);
```

Comment:
There's one little thing I'd like to change. I would move all this into `addDynamicTemplatesMapping()`, so we only have one call to this method and pass the XContentBuilder and class.

```java
private static void addDynamicTemplatesMapping(XContentBuilder xContentBuilder, Class claszz) throws IOException
```

Reply:
Good one. Applied refactoring.